packages/lit-html/src/lit-html.ts

```diff
@@ -512,10 +525,18 @@ const getTemplateHtml = (
     let attrName: string | undefined;
     let lastIndex = 0;
     let match!: RegExpExecArray | null;
-
     // The conditions in this loop handle the current parse state, and the
     // assignments to the `regex` variable are the state transitions.
     while (lastIndex < s.length) {
+      // If we are moving outside of a tag that should not contains
+      // internal parts, then reset that warning state.
+      if (
+        DEV_MODE &&
+        warnPartsInsideEndRegex! &&
+        s.match(warnPartsInsideEndRegex!)
```

Comment:
Because templates can nest, technically this would break since the tracking isn't balanced, right?

```js
html`<template><template></template>${foo}</template>`
```

Reply:
Seems like we'd need a stack, which is starting to get pretty fancy... Maybe timebox and just leave a TODO/issue if there's not an easy fix? I'm not sure we should spend a lot of time on this since it's just a DEV_MODE issue for a pretty corner case.